code/backend/batch/utilities/helpers/llm_helper.py

```diff
@@ -154,3 +156,34 @@ def get_sk_service_settings(self, service: AzureChatCompletion):
                 max_tokens=self.llm_max_tokens,
             ),
         )
+
+    def get_ml_client(self):
+        if not hasattr(self, "_ml_client"):
+            self._ml_client = MLClient(
+                DefaultAzureCredential(),
+                self.env_helper.AZURE_SUBSCRIPTION_ID,
+                self.env_helper.AZURE_RESOURCE_GROUP,
+                self.env_helper.AZURE_ML_WORKSPACE_NAME,
+            )
+        return self._ml_client
+
+    def get_endpoint_name(self):
+        return self.env_helper.PROMPT_FLOW_ENDPOINT_NAME
+
+    def get_deployment_name(self):
+        return self.env_helper.PROMPT_FLOW_DEPLOYMENT_NAME
+
+    def transform_chat_history_for_pf(self, chat_history):
+        transformed_chat_history = []
+        for i in range(0, len(chat_history), 2):
+            user_message = chat_history[i]['content']
+            assistant_message = chat_history[i+1]['content'] if i+1 < len(chat_history) else ''
```

Comment:
There can be scenarios where there are multiple user messages before an assistant message. Usually when there is an error. Can we get the message based on the role field? Maybe we can just ignore any user messages that dont' have an assistant response?

Reply:
good point, fixed as suggested